presto-main/src/main/java/com/facebook/presto/sql/planner/plan/JoinNode.java

```diff
@@ -100,6 +104,11 @@ public JoinNode(@JsonProperty("id") PlanNodeId id,
 
         checkArgument(!(criteria.isEmpty() && leftHashSymbol.isPresent()), "Left hash symbol is only valid in an equijoin");
         checkArgument(!(criteria.isEmpty() && rightHashSymbol.isPresent()), "Right hash symbol is only valid in an equijoin");
+
+        this.spatialJoin = criteria.isEmpty() && type == INNER && filter.isPresent()
+                && distributionType.map(REPLICATED::equals).orElse(false)
+                && extractSupportedSpatialFunctions(filter.get()).stream()
+                        .anyMatch(function -> spatialFunctionArgumentsMatchJoinNodes(function, left, right));
```

Comment:
This looks like some planner logic being embedded in the plan node. Isn't this  _the_ place where we make a decision whether join should be planned as spatial?